Libplanet/Net/Swarm.BlockSync.cs

```diff
@@ -16,436 +20,556 @@ public partial class Swarm<T>
         /// It is null when the <see cref="Swarm{T}"/> does not have any block to demand.
         /// <seealso cref="BlockDemand"/>
         /// </summary>
-        public BlockDemand? BlockDemand { get; private set; }
+        public ConcurrentDictionary<BoundPeer, BlockDemand> BlockDemands { get; private set; }
 
         internal AsyncAutoResetEvent FillBlocksAsyncStarted { get; } = new AsyncAutoResetEvent();
 
         internal AsyncAutoResetEvent FillBlocksAsyncFailed { get; } = new AsyncAutoResetEvent();
 
         internal AsyncAutoResetEvent ProcessFillBlocksFinished { get; } = new AsyncAutoResetEvent();
 
-        private async Task ProcessFillBlocks(
+        /// <summary>
+        /// Fill blocks from the <see cref="BoundPeer"/>s in the
+        /// <see cref="Swarm{T}.RoutingTable"/>.
+        /// </summary>
+        /// <param name="timeout">
+        /// The timeout value for the request to get the tip of the block.
+        /// </param>
+        /// <param name="maximumPollPeers">The maximum targets to send request to.</param>
+        /// <param name="cancellationToken">
+        /// A cancellation token used to propagate notification that this
+        /// operation should be canceled.</param>
+        /// <returns>An awaitable task without value.</returns>
+        public async Task ProcessFillBlocksAsync(
```

Comment:
Why has this changed to `public`? From the looks of it, I think it makes more sense to have `ProcessFillBlocksAsync()` as private and `FillBlocksAsync()` as `public` instead.

Reply:
`FillBlocksAsync()` is a task polls blocks by calling `ProcessFillBlocksAsync()` periodically. Do you think we should make polling task to public instead of actual function?

Reply:
I don't think it's a cut and dried issue. Multiple asynchronous calls to `ProcessFillBlocksAsync()` and/or `FillBlocksAsync()` don't seem to make sense anyway. So if it is all the same, if we were to expose a syncing method, I think more automated higher level function is the way to go. 🙃

Reply:
I lowered its access level to `internal`.